Sources/Vexil/Flag.swift

```diff
@@ -101,46 +176,53 @@ public struct Flag<Value>: Decorated, Identifiable where Value: FlagValue {
     ///                         You can also specify `.hidden` to hide this flag from Vexillographer.
     ///
     public init (wrappedValue: Value, name: String? = nil, codingKeyStrategy: CodingKeyStrategy = .default, description: FlagInfo) {
-        self.codingKeyStrategy = codingKeyStrategy
-        self.defaultValue = wrappedValue
-
         var info = description
         info.name = name
-        self.info = info
+        self.allocation = Allocation(
+            info: info,
+            defaultValue: wrappedValue,
+            codingKeyStrategy: codingKeyStrategy
+        )
     }
 
 
     // MARK: - Decorated Conformance
 
-    internal var decorator = Decorator()
-    internal let codingKeyStrategy: CodingKeyStrategy
-
     /// Decorates the receiver with the given lookup info.
     ///
     /// `self.key` is calculated during this step based on the supplied parameters. `lookup` is used by `self.wrappedValue`
     /// to find out the current flag value from the source hierarchy.
     ///
-    internal func decorate (lookup: Lookup, label: String, codingPath: [String], config: VexilConfiguration) {
-        self.decorator.lookup = lookup
-
-        var action = self.codingKeyStrategy.codingKey(label: label)
+    internal func decorate (
+        lookup: Lookup,
+        label: String,
+        codingPath: [String],
+        config: VexilConfiguration
+    ) {
+        // FIXME: this doesn't use `isKnownUniquelyReferenced`, but perhaps it should?
+        self.allocation.lookup = lookup
+
+        var action = self.allocation.codingKeyStrategy.codingKey(label: label)
         if action == .default {
             action = config.codingPathStrategy.codingKey(label: label)
         }
 
         switch action {
 
         case .append(let string):
-            self.decorator.key = (codingPath + [string])
+            // FIXME: for compatibility with existing behavior, this doesn't use `isKnownUniquelyReferenced`, but perhaps it should?
```

Comment:
Fixme is asking a question, not proposing something to be fixed in the future. Can we just answer the question now? `decorate` should only be called once during the FlagPole's init. If that warrants `isKnownUniquelyReferenced` and would be a breaking change then lets change this into a statement + GitHub issue so we can track it.

Reply:
I've removed these comments. I believe the new behavior is equivalent to the existing behavior. Whether that was already wrong in edge cases is beyond the scope of this PR 😅 (but from what you say above, I don't think it matters)